src/System.Runtime/tests/System/Array.cs

```diff
@@ -1068,6 +1094,22 @@ public static void TestLastIndexOf()
     }
 
     [Fact]
+    public static void TestLastIndexOf_Invalid()
+    {
+        Assert.Throws<ArgumentNullException>(() => Array.LastIndexOf((int[])null, 1, 0, 0));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.LastIndexOf(new int[] { }, 1, 1, 1));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.LastIndexOf(new int[] { }, 1, -1, 1));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.LastIndexOf(new int[] { }, 1, 0, 1));
+        Assert.Equal(-1, Array.LastIndexOf(new int[] { }, 1, -1, 0));
+        Assert.Equal(-1, Array.LastIndexOf(new int[] { }, 1, 0, 0));
```

Comment:
It is a bit strange. It follows a special code path that wasn't being covered before where the index is {-1,0} and the count is 0 it returns -1. I'll move it to a special-case test method.